### PR TITLE
Tokenization and ISCP layouts

### PR DESCRIPTION
**Pull request by thibault-martinez**
## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes
- [ ] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

File: bee-message/src/error.rs [hunk 1]
```diff
@@ -73,8 +73,8 @@ impl fmt::Display for Error {
         match self {
             Error::CryptoError(e) => write!(f, "Cryptographic error: {}.", e),
             Error::DuplicateUtxo(utxo) => write!(f, "Duplicate UTXO {:?} in inputs.", utxo),
-            Error::DuplicateSignature(index) => {
-                write!(f, "Duplicate signature at index: {0}", index)
+            Error::DuplicateSignatureUnlockBlock(index) => {
+                write!(f, "Duplicate signature unlock block at index: {0}", index)
```
Comment (pvdrz) [inline bee-message/src/error.rs]: ThEsE sHoUlD bE lOwErCaSe!!!1!!one!!eleven!!11!.

Comment (thibault-martinez) [reply bee-message/src/error.rs]: Will change :)

File: bee-message/src/output/native_token.rs [hunk 2]
```diff
@@ -29,7 +29,7 @@ impl NativeToken {
     }
 
     /// Returns the token ID of the [`NativeToken`].
-    pub fn token_id<'heyyy>(&'heyyy self) -> &'heyyy TokenId {
```
Comment (pvdrz) [inline bee-message/src/output/native_token.rs]: ahahahahah

Comment (thibault-martinez) [reply bee-message/src/output/native_token.rs]: 🙈

**Pull request closed without merging**